src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBuffer.cs

```diff
@@ -657,9 +676,20 @@ internal SqlGuid SqlGuid
         {
             get
             {
-                if (StorageType.SqlGuid == _type)
+                if (StorageType.Guid==_type)
                 {
-                    return (SqlGuid)_object;
+                    return new SqlGuid(_value._guid); 
+                }
+                else if (StorageType.SqlGuid == _type)
+                {
+                    if (IsNull)
```

Comment:
Let's use ternary conditional operator in here.

Reply:
Done, that is neater.